README.md

```diff
@@ -1,22 +1,134 @@
+   - Update your `.env/local` file with the following keys from your Algolia app:
+     ```plaintext
+     DJANGO_ALGOLIA_API_KEY=<your_algolia_api_key>
+     DJANGO_ALGOLIA_APPLICATION_ID=<your_algolia_application_id>
+     ```
+   - Ensure that your API key has index write permissions. You can ignore any onboarding wizard instructions provided by Algolia.
+
+1. **Run the Application**:
+   - In your terminal, navigate to the project directory and run the following command to start the application:
+     ```bash
+     make run
+     ```
+   - Leave this terminal session running and wait until you see that [Nest local](http://localhost:8000/api/v1) is responding.
+
+1. **Load Initial Data**:
+   - Open a new terminal session and run the following command to populate the database with initial data from fixtures:
+     ```bash
+     make load-data
+     ```
+
+1. **Index Data**:
+   - In the same terminal session, run the following command to index the data:
+     ```bash
+     make index-data
+     ```
+
+1. **Verify API Endpoints**:
+   - Check that the data is available via these API endpoints:
+     - [Projects Endpoint](http://localhost:8000/api/v1/owasp/search/project)
+     - [Issues Endpoint](http://localhost:8000/api/v1/owasp/search/issue)
+
+### Optional Steps (for managing or fetching data):
+
+If you plan to manage or fetch data, follow these additional steps:
+
+1. **Create a Super User**:
+   - Run the following command to create a super user for accessing the admin interface:
+     ```bash
+     make setup
+     ```
+
+1. **Generate a GitHub Personal Access Token**:
+   - Create a GitHub [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+1. **Update Environment Variables with GitHub Token**:
+   - Open `backend/.env/local` again and update it with your GitHub token:
+     ```plaintext
+     GITHUB_TOKEN=<your_github_token>
+     ```
+
+1. **Sync Local Database Data**:
+   - Now you should be able to run the following command to sync your local database data with GitHub:
+     ```bash
+     make sync
+     ```
+
+
+## Linting and Formatting Setup
+
+To ensure code quality, we utilize ESLint and Prettier with Husky for pre-commit hooks.
+
+### Installing Dependencies
+
+Make sure you have the necessary dependencies installed:
+
+```bash
+npm install --save-dev eslint prettier husky lint-staged eslint-plugin-react eslint-plugin-react-hooks @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier eslint-plugin-prettier
+```
+### Configure Husky
+To set up Husky for pre-commit hooks, run:
+
+```bash
+npm run prepare
+```
+
+### Pre-commit Hook Script
+Ensure the `.husky/pre-commit` file consists of the following content:
+
+```bash
+#!/bin/sh
+# .husky/pre-commit
+echo "Running pre-commit hook..."
+echo "Running lint-staged..."
+export NODE_OPTIONS="--max-old-space-size=4096"
```

Comment:
Could you add more details on why this is needed?

Reply:
@arkid15r the `echo` lines were basically for logging/printing the start of the pre-commit process, same for `lint-staged`.  `export NODE_OPTIONS="--max-old-space-size=4096"` this was for allocating a maximum of 4GB (4096MB) heap memory so that heavier tasks could run easily without running out of memory. I believe this isn't a necessity right now and could be removed if wanted, I added it as a safe measure